# Pengenalan Dataset

Disini kamu akan melakukan elaborasi untuk persiapan data sebelum dilakukan analisis. Adapun dataset yang digunakan adalah **North wind db** yang terdiri dalam berikut ini:

1. dataset order_details terdiri: **order_id, product_id, unit_price, quantity, discount**
2. dataset orders: **order_id, customer_id, employee_id, order_date, required_date, shipped_date, ship_via, freight, ship_name, ship_address, ship_city, ship_region,	ship_postal_code,	ship_country,	born_year**
3. dataset products: **product_id, product_name, supplier_id, category_id,	quantity_per_unit,	unit_price,	units_in_stock, units_on_order, reorder_level, discontinued**
4. dataset customers: **customer_id, company_name, contact_name,	contact_title,	address,	city,	region,	postal_code,	country,	phone,	fax**

Data tersebut dapat dilakukan penarikan dengan proses dibawah ini:

```python
import psycopg2 as pg
import pandas.io.sql as psql
import pandas as pd

# Bangun konektivitas
host = "ec2-54-164-22-242.compute-1.amazonaws.com"
dbname = "d92sarup3qlji0"
user = "fttxsdmckdfikz"
password = "c5561e667cd292f382a0e437807e05c89fba1b894dddff1e77e07befe8a866ac"
connection = pg.connect(f"host={host} dbname={dbname} user={user} password={password}")

# Function SQL
def sql(doc):
    return psql.read_sql_query(doc, connection)
```



In [ ]:
# Silahkan dicoba disini



In [ ]:
# Berapakah jumlah kota yang dituju dalam export?
sql("""

""")

total
0     70

In [ ]:
sql("""

""")

## Langkah 1

Seperti kita ketahui bahwa jumlah transaksi yang terjadi dalam order_details mencapai 2155 transaksi. Berdasarkan data tersebut, negara mana yang memiliki transaksi terbesar dan terkecil?

In [ ]:
## Silahkan tuliskan disini
# 2 MAXIMUM
sql("""

""")

ship_country  data_count
0          USA         352

In [ ]:
# 2 MINIMUM
sql("""

""")

ship_country  data_count
0       Norway          16

Kita ketahui bahwa USA memiliki transaksi terbesar sebanyak 352 transaksi dan Norway transaksi terkecil sebanyak 16 transaksi.

## Langkah 2

Seperti kita ketahui bahwa jumlah transaksi yang terjadi dalam order_details mencapai 2155 transaksi. Berdasarkan data tersebut, produk apa yang memiliki transaksi terbesar dan terkecil?

In [ ]:
## Silahkan tuliskan disini
# 3 MAXIMUM
sql("""

""")

product_name  data_count
0  Raclette Courdavault          54

In [ ]:
# 3 MINIMUM
sql("""

""")

product_name  data_count
0  Mishi Kobe Niku           5

Kita ketahui bahwa Raclette Courdavault memiliki transaksi terbesar sebanyak 54 transaksi dan Mishi Kobe Niku transaksi terkecil sebanyak 5 transaksi.

## Langkah 3

Berdasarkan informasi langkah 1, produk apa saja yang banyak diexport ke USA dan Produk Raclette Courdavault banyak dilakukan transaksi kemana saja?

In [ ]:
# 4 PRODUCT HITS IN USA
sql("""

""")

product_name  total
0  Gnocchi di nonna Alice     17

In [ ]:
# 5 DIMANA LOKASI EXPORT
sql("""

""")

ship_country  total
0      Germany     11

## Langkah 4

Buatlah dataframe yang mengandung feature: 
1. order_date, 
2. freight, 
3. ship_via, 
4. unit_price, 
5. quantity, 
6. discount, 
7. reorder_level, 
8. units_in_stock, 
9. units_in_order
10. country

In [ ]:
# 6 Nomor
df = sql("""

""")
df.head()

order_date  freight  ship_via  unit_price  quantity  discount  \
0  1996-07-04    32.38         3        14.0        12       0.0   
1  1996-07-04    32.38         3         9.8        10       0.0   
2  1996-07-04    32.38         3        34.8         5       0.0   
3  1996-07-05    11.61         1        18.6         9       0.0   
4  1996-07-05    11.61         1        42.4        40       0.0   

   reorder_level  units_in_stock  units_on_order  country  
0             30              22              30   France  
1              0              26               0   France  
2              0              14               0   France  
3              0              35               0  Germany  
4             10              20               0  Germany

## Langkah 5

Lakukan profiling dengan menampilkan
1. Missing Value
2. Informasi Statistik
3. Korelasi Data

In [ ]:
# 7 Check missing value
df

order_date        False
freight           False
ship_via          False
unit_price        False
quantity          False
discount          False
reorder_level     False
units_in_stock    False
units_on_order    False
country           False
dtype: bool

In [ ]:
# 7 Check informasi statistik
for i in df.select_dtypes(include=['int', 'float']).columns:
  print("Informasi statistik dari " + )
  print("Minimum value:", )
  print("Maximum value:", )
  print("Mean value:", )
  print("Median value:", )
  print("Std value:", )

Informasi statistik dari freight
Minimum value: 0.02
Maximum value: 1007.64
Mean value: 96.19772621809764
Median value: 53.8
Std value: 133.01127192559946
Informasi statistik dari ship_via
Minimum value: 1
Maximum value: 3
Mean value: 1.9995359628770302
Median value: 2.0
Std value: 0.774176873198129
Informasi statistik dari unit_price
Minimum value: 2.0
Maximum value: 263.5
Mean value: 26.21851972157779
Median value: 18.4
Std value: 29.827417887501035
Informasi statistik dari quantity
Minimum value: 1
Maximum value: 130
Mean value: 23.812993039443157
Median value: 20.0
Std value: 19.022046977493545
Informasi statistik dari discount
Minimum value: 0.0
Maximum value: 0.25
Mean value: 0.05616705336426916
Median value: 0.0
Std value: 0.08345026112646185
Informasi statistik dari reorder_level
Minimum value: 0
Maximum value: 30
Mean value: 11.900232018561486
Median value: 10.0
Std value: 10.791420598550184
Informasi statistik dari units_in_stock
Minimum value: 0
Maximum value: 125
Mean value

In [ ]:
# 7 Check korelasi
df.select_dtypes(include=['int', 'float'])

freight  ship_via  unit_price  quantity  discount  \
freight         1.000000  0.050721    0.241306  0.370832  0.023498   
ship_via        0.050721  1.000000    0.016892 -0.013940 -0.040197   
unit_price      0.241306  0.016892    1.000000  0.007366 -0.011166   
quantity        0.370832 -0.013940    0.007366  1.000000  0.128577   
discount        0.023498 -0.040197   -0.011166  0.128577  1.000000   
reorder_level  -0.039750 -0.022400   -0.150803  0.008573  0.017835   
units_in_stock -0.066197 -0.007188   -0.174785  0.016473 -0.003835   
units_on_order  0.002363 -0.010990   -0.115969  0.035448  0.010167   

                reorder_level  units_in_stock  units_on_order  
freight             -0.039750       -0.066197        0.002363  
ship_via            -0.022400       -0.007188       -0.010990  
unit_price          -0.150803       -0.174785       -0.115969  
quantity             0.008573        0.016473        0.035448  
discount             0.017835       -0.003835        0.010167  
reorder_level        1.000000        0.246886        0.361687  
units_in_stock       0.246886        1.000000       -0.331962  
units_on_order       0.361687       -0.331962        1.000000

## Langkah 6

Lakukan cleansing dengan
1. Atasi missing value
2. Atasi outlier
3. Atasi duplikasi


In [ ]:
# 8 Tuliskan disini


